File: src/main/java/com/elf/vipForElf/domain/building/vo/BuildingListVO.java
package com.elf.vipForElf.domain.building.vo;

import com.elf.vipForElf.domain.building.entity.BuildingEntity;
import lombok.Value;

@Value
public class BuildingListVO {
    String buildingName;
    String buildingNumber;

    public BuildingListVO(BuildingEntity buildingEntity){
        this.buildingName = buildingEntity.getBuildingName();
        this.buildingNumber = buildingEntity.getBuildingNumber();
    }
}

File: src/main/java/com/elf/vipForElf/domain/building/vo/BuildingListVO.kt
package com.elf.vipForElf.domain.building.vo

import com.elf.vipForElf.domain.building.entity.BuildingEntity

data class BuildingListVO(
    val buildingName: String,
    val buildingNumber: String
) {
    constructor(buildingEntity: BuildingEntity) : this(
        buildingName = buildingEntity.buildingName,
        buildingNumber = buildingEntity.buildingNumber ?: ""
    )
}
